### DISCUSSION TITLE
Changing font size

### DISCUSSION THREAD

**Discussion by toorajpro** [arthelokyo/astrowind#398] [Q&A] [answered]
Hi,

It seems the current font size for normal paragraphs is too big (25px). Is there a way to reduce all the font sizes with the same scale factor? For example I like to have the paragraph font to be reduced to 22px so a reduction ratio if 22/25, or 88%. How can I achive a 88% reduction on all the font sizes? Also how to change the line spacing in the paragraphs?

**Comment by prototypa** [comment, marked as answer]
Hi @toorajpro


I think a solution for your needs is to modify the [tailwind.config.cjs](https://github.com/onwidget/astrowind/blob/main/tailwind.config.cjs) to adjust the Tailwind CSS settings to update the font sizes and line height to the values ​​you want.

- https://tailwindcss.com/docs/font-size#customizing-your-theme

Also keep in mind that for larger devices in **AstroWind** we do a proportional adjustment (increase the size of everything) using a class in the HTML tag. 

```html
<html class="2xl:text-[20px] motion-safe:scroll-smooth">
  <!-- rest of the tags -->
</html>
```

The class `2xl:text-[20px]` means that the base font for devices with a minimum size of 1536px will be 20px. More details:

- https://tailwindcss.com/docs/responsive-design

**Reply by toorajpro** [reply]
Thanks for the answer, it was very helpful.

However, I tried changing the base font from 20px, which is enforced for devices with a minimum size of 1536 px as you explained, to 18px in the `Layout.astro` file as

`<html lang={language} dir={textDirection} class="2xl:text-[20px]">`

But it stayed at 20px and didn't change on the website. Do I need to change something else as well?
I must mention that per your first link I also reduced the font sizes in `tailwind.config.cjs` as follows:

```
      fontSize: {
        sm: '0.8rem',
        base: '0.9rem',
        xl: '1.0rem',
        '2xl': '1.25rem',
        '3xl': '1.5rem',
        '4xl': '2.0rem',
        '5xl': '3.0rem',
      },
```

Thanks again for the help.

**Reply by prototypa** [reply]
Hi @toorajpro 

Try updating `Layout.astro` for this line and tell me if it works for you, I have exaggerated the values ​​so the change is noticeable

```html
<html lang={language} dir={textDirection} class="text-[6px] 2xl:text-[10px]">
```

That will proportionally update all theme fonts using the base font specified in the `<html>` tag.

But if you want to change only the font size of some variations then you can do it with the `tailwind.config.js` configuration

**Reply by toorajpro** [reply]
Thanks again,

What you suggested reduces font sizes. But it it too small. If instead of [10px] I use [18px] then the fonts on blog posts are at size 23px and the font on the main page just under the website title is 18px. How can I change my blog post font size to 18px or 20px and keep the font size on the main page unchanged?

**Reply by prototypa** [reply]
I exaggerated the numbers to see if the change was noticeable. I see it works for you.

Now, about the Blog Post, the thing there is that by default we decided to use a larger font than normal to make the content stand out. That is configured in this line:

https://github.com/onwidget/astrowind/blob/main/src/components/blog/SinglePost.astro#L91

```html
<div
      class="mx-auto px-6 sm:px-6 max-w-3xl prose prose-md lg:prose-xl dark:prose-invert dark:prose-headings:text-slate-300 prose-headings:font-heading prose-headings:leading-tighter prose-headings:tracking-tighter prose-headings:font-bold prose-a:text-primary dark:prose-a:text-blue-400 prose-img:rounded-md prose-img:shadow-lg mt-8 prose-headings:scroll-mt-[80px]"
    >
```

All those classes starting with prose are from the official Tailwind CSS Typography plugin https://tailwindcss.com/docs/plugins#typography

You have these classes modifying the font size of the articles: `prose prose-md lg:prose-xl`
Try changing them to other values ​​specified by the plugin
